sign_oca/static/src/components/sign_oca_pdf_portal/sign_oca_navigator.esm.js

```diff
@@ -0,0 +1,193 @@
+/** @odoo-module QWeb **/
+/* global document, window, Event, KeyboardEvent */
+import {_t} from "web.core";
+
+export function offset(el) {
+    const box = el.getBoundingClientRect();
+    const docElem = document.documentElement;
+    return {
+        top: box.top + window.scrollY - docElem.clientTop,
+        left: box.left + window.scrollY - docElem.clientLeft,
+    };
+}
+
+/**
+ * Starts the sign item navigator
+ * @param { SignablePDFIframe } parent
+ * @param { HTMLElement } target
+ * @param { Environment } env
+ */
+export function startSignItemNavigator(parent, target, env) {
+    const state = {
+        started: false,
+        isScrolling: false,
+    };
+    const navigator = parent.iframe.el.contentDocument.getElementsByClassName(
+        "o_sign_sign_item_navigator"
+    )[0];
+    const navLine = parent.iframe.el.contentDocument.getElementsByClassName(
+        "o_sign_sign_item_navline"
+    )[0];
+    const checkSignItemsCompletion = parent.checkSignItemsCompletion();
+    let signItemsToComplete = checkSignItemsCompletion;
+
+    function _scrollToSignItemPromise(item) {
+        return new Promise((resolve) => {
```

Comment:
I think you should add error handling in `Promises`

```suggestion
        return new Promise((resolve, reject) => {
```